qualification/cbf.py

```diff
@@ -130,38 +124,24 @@ async def steady_state_timestamp(self, *, max_delay: int = DEFAULT_MAX_DELAY) ->
         acceptable for most cases.
         """
         timestamp = 0
-        # Although the dsim sensors will also appear in the product controller,
-        # we can't rely on that due to a race condition: if we make a change
-        # directly on the dsim, the subscription update it sends to the product
-        # controller might not be received before we ask the product controller
-        # for the sensor value. So we have to query every device server that we
-        # make state changes though.
-        clients = [self.product_controller_client] + self.dsim_clients
-        async with asyncio.TaskGroup() as tg:
-            requests = [
-                tg.create_task(client.request("sensor-value", r"/.*steady-state-timestamp$/")) for client in clients
-            ]
-        for client, request in zip(clients, requests):
-            _, informs = request.result()
-            for inform in informs:
-                # In theory there could be multiple sensors per inform, but aiokatcp
-                # never does this because timestamps are seldom shared.
-                sensor_value = int(inform.arguments[4])
-                if client is not self.product_controller_client:
-                    # values returned from the dsim do not account for delay,
-                    # so need to be offset to get an output timestamp.
-                    sensor_value += max_delay
-                timestamp = max(timestamp, sensor_value)
+        _, informs = await self.product_controller_client.request("sensor-value", r"/.*steady-state-timestamp$/")
+        for inform in informs:
+            # In theory there could be multiple sensors per inform, but aiokatcp
+            # never does this because timestamps are seldom shared.
+            sensor_name = inform.arguments[2]  # Byte string
+            sensor_value = int(inform.arguments[4])
```

Comment:
I'm not quite following this. Probably easiest to discuss via Zoom. Will drop you a message now.